src/google/protobuf/arena.h

```diff
@@ -258,7 +258,7 @@ class PROTOBUF_EXPORT PROTOBUF_ALIGNAS(8) Arena final {
   template <typename T>
   PROTOBUF_NDEBUG_INLINE static T* PROTOBUF_NONNULL
   CreateArray(Arena* PROTOBUF_NULLABLE arena, size_t num_elements) {
-    static_assert(std::is_trivial<T>::value,
+    static_assert(std::is_trivially_default_constructible<T>::value,
```

Comment:
For this one, this is effectively a revert of a 4 years old commit citing "old libstdc++". I hope it's ok now: https://github.com/protocolbuffers/protobuf/commit/b74892e9c368ee3517c404b569b3b447006518b6

Reply:
Yea this seems reasonable to me, especially since we never even updated the error message :P. I think this effectively allows types that aren't trivially copyable now, but I don't think that's actually a requirement for arena allocation